server/controllers/content/contentDelete.ts

```diff
@@ -0,0 +1,118 @@
+import { logger } from "@project-sunbird/ext-framework-server/logger";
+import { Manifest } from "@project-sunbird/ext-framework-server/models";
+import * as childProcess from "child_process";
+import * as _ from "lodash";
+import { containerAPI } from "OpenRAP/dist/api";
+import * as path from "path";
+import { Inject } from "typescript-ioc";
+import DatabaseSDK from "../../sdk/database";
+import Response from "../../utils/response";
+import Content from "./content";
+let deleted = [];
+const failed = [];
+export default class ContentDelete {
+    public contentPath;
+    private workerProcessRef: childProcess.ChildProcess;
+    @Inject
+    private databaseSdk: DatabaseSDK;
+    private fileSDK: any;
+    private content: Content;
+    constructor(manifest: Manifest) {
+        this.content = new Content(manifest);
+        this.databaseSdk.initialize(manifest.id);
+        this.fileSDK = containerAPI.getFileSDKInstance(manifest.id);
+        this.contentPath = [];
+    }
+
+    public async delete(req, res) {
+        const reqId = req.headers["X-msgid"];
+        logger.debug(`${reqId}: Delete method is called`);
+        this.workerProcessRef = childProcess.fork(path.join(__dirname, "contentDeleteHelper"));
```

Comment:
if the process is already running still this creates a new one for each API call